server/src/main/java/org/opensearch/extensions/rest/RestSendToExtensionAction.java

```diff
@@ -259,7 +250,9 @@ public String executor() {
                     filteredHeaders,
                     contentType,
                     content,
-                    requestIssuerIdentity,
+                    identityService.getTokenManager()
+                        .issueOnBehalfOfToken(List.of(discoveryExtensionNode.getId(), identityService.getSubject().toString()))
```

Comment:
Only the extension unique id is needed here to populate the `aud` of the token.

Within the implementation of `issueOnBehalfOfToken` it already has access to the currently authenticated user and should use the username of that subject as the `sub` claim.

Reply:
Oh I see, I thought you wanted both the User and and the id. I will swap that.